### PR TITLE
Implement for par_iter and par_iter_mut on Query

### PR DESCRIPTION
**Pull request by TheRawMeatball**

### CODE REVIEW

**Comment by Frizi** [comment]
Shouldn't that variant be called `par_map`?

**Comment by TheRawMeatball** [comment]
Alright, I rewrote the whole thing, and now it exposes par_iter methods on queries which let you use the parallel iterator interface in bevy_tasks.

File: crates/bevy_tasks/src/single_threaded_task_pool.rs [hunk 1]
```diff
@@ -59,7 +59,7 @@ impl TaskPool {
     /// This is similar to `rayon::scope` and `crossbeam::scope`
     pub fn scope<'scope, F, T>(&self, f: F) -> Vec<T>
     where
-        F: FnOnce(&mut Scope<'scope, T>) + 'scope + Send,
+        F: FnOnce(&mut Scope<'scope, T>) + 'scope,
```
Comment (bjorn3) [inline crates/bevy_tasks/src/single_threaded_task_pool.rs:62]: This makes the single threaded and multi threaded task pool not have the same function signature.

Comment (TheRawMeatball) [reply crates/bevy_tasks/src/single_threaded_task_pool.rs:62]: No? I removed this unnecessary Send bound on both versions.

Comment (bjorn3) [reply crates/bevy_tasks/src/single_threaded_task_pool.rs:62]: Right, my bad. Looked at a single commit.

**Comment by CGMossa** [comment]
I find myself in need of this PR. I have many places where I am filtering, and collecting based on query-data, and I wish to make these parallel executed.
If I can help with this PR, then that would be great.

**Comment by ManevilleF** [comment]
What is the status of this MR ?

**Comment by TheRawMeatball** [comment]
I'll fix it up eventually, though probably not that soon, given it's likely to break again.

**Comment by alice-i-cecile** [comment]
@TheRawMeatball, are you interested in cleaning up the PR description and rebasing? Or should this wait on future work.

Totally fine if it's not a priority, but I'll add a `Abandoned` label so other folks know to pick up the work. This would be valuable, so I don't want to leave it sitting forever.

**Comment by ManevilleF** [comment]
I think this PR could be very valuable, it can avoid using `Mutex` or `RwLock` when extracting parallelized block data

**Comment by mockersf** [comment]
[`Query::par_iter`](https://docs.rs/bevy/0.10.1/bevy/prelude/struct.Query.html#method.par_iter) and [`Query::par_iter_mut`](https://docs.rs/bevy/0.10.1/bevy/prelude/struct.Query.html#method.par_iter_mut) now exists

**Pull request closed without merging**